src/v/serde/serde.h

```diff
@@ -279,8 +286,6 @@ void write(iobuf& out, T t) {
         return write(out, static_cast<typename Type::type>(t));
     } else if constexpr (reflection::is_ss_bool_v<Type>) {
         write(out, static_cast<int8_t>(bool(t)));
-    } else if constexpr (std::is_same_v<Type, std::chrono::milliseconds>) {
```

Comment:
>In https://github.com/redpanda-data/redpanda/pull/5427 @BenPope rightly pointed that this is not backward compatible since we change the on disk/wire serialization but @dotnwat mentioned that no one may be using this (yet) given the framework is fairly new (need to verify).

correct. this needs to be researched and verified, but does seem like a viable path forward.

this needs to be done in a the next week or two before the release.

Reply:
I've gone thru all serde callers in `21.11.x` and `22.1.x`, only interesting I could find are `cloudstorage::partition_manifest::segment_meta` and  `storage::index_state`  and both of them use `model::timestamp `which treats timestamps as int64_t, so I don't think we are breaking any compat here.